### PR TITLE
fix(detectBrowserLanguage): use ISO code if available

### PR DESCRIPTION
**Pull request by owenandrews**
This PR updates `matchBrowserLocale` to use the ISO code for a locale when performing browser language detection. If the ISO code isn't set then the usual locale code is used, as is the current functionality.

This solves some of the issues mentioned in #499 where one may not want to use ISO codes in the URL, but does wish to make use of the browser detection and SEO features.

This is my first attempt at contributing to this project so I'm not 100% sure that the way I've gone about it is correct and I would welcome feedback on how this PR can be made better :)

### CODE REVIEW

File: src/templates/utils-common.js [hunk 1]
```diff
@@ -25,26 +25,29 @@ export const matchBrowserLocale = (appLocales, browserLocales) => {
   /** @type {{ code: string, score: number }[]} */
   const matchedLocales = []
 
+  // Normalise appLocales input
```
Comment (rchl) [inline src/templates/utils-common.js:28]: The types of `appLocales` in the JSDoc comment above should be updated.

**Review by rchl — COMMENTED** [review]
Please delete `package-lock.json` and use `yarn` in this project.
(Or keep using npm but don't commit the lock file.)

**Comment by rchl** [comment]
Sorry to keep you waiting. I haven't reviewed it properly yet, will do it ASAP, when I have more time.

**Comment by rchl** [comment]
This would need some new tests added around here: https://github.com/nuxt-community/i18n-module/blob/fee0e9375ef050e05ffc9a29d69fe0216fe50e03/test/module.test.js#L1166-L1192

Base config used by those tests is here: https://github.com/nuxt-community/i18n-module/blob/fee0e9375ef050e05ffc9a29d69fe0216fe50e03/test/fixture/base.config.js#L33-L44

**Comment by rchl** [comment]
Actually, tests can be unit-tested here: https://github.com/nuxt-community/i18n-module/blob/6bac69c1cd3ca51f0ab675460a63be4509d19c01/test/unit.test.js#L31-L31
It's easier and faster to work with those.

**Comment by owenandrews** [comment]
Thanks @rchl, I added a unit test, does it need to be comprehensive or is just the one ok in this instance?

File: src/templates/utils-common.js [hunk 2]
```diff
@@ -17,34 +17,37 @@ export const parseAcceptLanguage = input => {
...
  * @param {string[]} browserLocales The locales to match against configured.
  * @return {string?}
  */
 export const matchBrowserLocale = (appLocales, browserLocales) => {
   /** @type {{ code: string, score: number }[]} */
   const matchedLocales = []
 
+  // Normalise appLocales input
+  appLocales = appLocales.map(appLocale => ({
+    code: typeof appLocale === 'string' ? appLocale : appLocale.code,
```
Comment (rchl) [inline src/templates/utils-common.js:30]: We don't seem to be using the `code` property so why create it?

File: test/unit.test.js [hunk 3]
```diff
@@ -94,4 +94,11 @@ describe('matchBrowserLocale', () => {
 
     expect(matchBrowserLocale(appLocales, browserLocales)).toBe('en-gb')
   })
+
+  test('matches ISO locale code', () => {
+    const appLocales = [{ code: 'cn', iso: 'zh-CN' }, { code: 'en', iso: 'en-US' }]
+    const browserLocales = ['zh', 'zh-CN']
+
+    expect(matchBrowserLocale(appLocales, browserLocales)).toBe('cn')
+  })
```
Comment (rchl) [inline test/unit.test.js:103]: Let's also add something more tricky:

```
  test('matches full ISO code', () => {
    const appLocales = [{ code: 'us', iso: 'en-US' }, { code: 'gb', iso: 'en-GB' }]
    const browserLocales = ['en-GB', 'en']

    expect(matchBrowserLocale(appLocales, browserLocales)).toBe('gb')
  })
```

EDIT: And also this one (taken from another PR that is currently (that is without this fix) failing):
```
  test('matches locale when only languages match', () => {
    const appLocales = ['en-GB', 'en-US']
    const browserLocales = ['en-IN', 'en']

    expect(matchBrowserLocale(appLocales, browserLocales)).toBe('en-GB')
  })
```

And there is also this tricky case in which none of the matches is exact:

```
  test('matches locale when only languages match', () => {
    const appLocales = ['en-CA', 'fr-CA']
    const browserLocales = ['en-UK', 'fr']

    expect(matchBrowserLocale(appLocales, browserLocales)).toBe('en-CA')
  })
```

The expected result for this one is not entirely clear but let's have it test the current behavior anyway. Maybe add a comment to that effect?

**Review by rchl — APPROVED** [review]
Looks good to me. Just please update with this one extra test.

**Comment by rchl** [comment]
Can you also update the `options-reference.md` (https://i18n.nuxtjs.org/options-reference#locales) to mention that the `iso` property is also used for matching browser locales when using the `detectBrowserLanguage` option? If you do, please update both `en` and `es` pages (both with English text).

File: test/unit.test.js (hunk 3 above)
Comment (rchl) [reply test/unit.test.js:103]: Pushed myself.

**Comment by rchl** [comment]
Thanks!

**Comment by owenandrews** [comment]
Thanks @rchl, sorry you beat me to it.

**Pull request merged**